src/common/proc_macro/src/lib.rs

```diff
@@ -247,3 +248,10 @@ pub fn derive_estimate_size(input: TokenStream) -> TokenStream {
         }
     }
 }
+
+#[proc_macro_derive(Guc, attributes(guc_opts))]
+#[proc_macro_error]
+pub fn guc(input: TokenStream) -> TokenStream {
```

Comment:
Would you mind adding documentation to the derive macro? For example, I didn't find what `flag` is for, while others are intuitive.